net/odp.go

```diff
@@ -0,0 +1,135 @@
+package net
+
+import (
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/weaveworks/go-odp/odp"
+)
+
+// ODP admin functionality
+
+func CreateDatapath(dpname string, mtuToCheck int) (supported bool, validMTU bool, err error) {
+	validMTU = true
+
+	dpif, err := odp.NewDpif()
+	if err != nil {
+		if odp.IsKernelLacksODPError(err) {
+			return false, validMTU, nil
+		}
+		return true, validMTU, err
+	}
+	defer dpif.Close()
+
+	dp, err := dpif.CreateDatapath(dpname)
+	if err != nil && !odp.IsDatapathNameAlreadyExistsError(err) {
+		return true, validMTU, err
+	}
+
+	var (
+		vpid      odp.VportID
+		vpname    string
+		errCreate error
+	)
+	if vpid, vpname, errCreate = createDummyVxlanVport(dp); err != nil {
```

Comment:
I don't think so. If `vxlan_vport` kernel module is not present, the check will fail before trying to open any port.